app_build_suite/build_steps/helm.py

```diff
@@ -333,19 +335,20 @@ def run(self, config: argparse.Namespace, context: Dict[str, Any]) -> None:
             logger.info(str(line, "utf-8"))
             if line.startswith(b"Successfully packaged chart and saved it to"):
                 full_chart_path = str(line.split(b":")[1].strip(), "utf-8")
+                full_chart_path = os.path.abspath(full_chart_path)
                 # compare our expected chart_file_name with the one returned from helm and fail if differs
                 helm_chart_file_name = os.path.basename(full_chart_path)
                 if helm_chart_file_name != context[context_key_chart_file_name]:
                     raise BuildError(
                         self.name,
                         f"unexpected chart path '{helm_chart_file_name}' != '{context[context_key_chart_file_name]}'",
                     )
-                if not context[context_key_chart_full_path].endswith(full_chart_path):
+                if full_chart_path != context[context_key_chart_full_path]:
                     raise BuildError(
-                        self.name, f"expected '{context[context_key_chart_full_path]}' to end with '{full_chart_path}'"
+                        self.name,
+                        f"unexpected helm build result: path reported in output '{full_chart_path}' "
+                        f"is not equal to '{context[context_key_chart_full_path]}'",
                     )
```

Comment:
Did you test this with `-c` values other than `.`?

Reply:
Yes, it works now. I reviewed the code and every property in `context` is now set only once.